operator/pkg/utils/secrets/secrets.go

```diff
@@ -26,21 +26,41 @@ import (
 
 type Request struct {
 	*certutil.Config
+	CASecret  *v1.Secret
+	Type      RequestType
 	Name      string
 	Namespace string
 }
 
+type RequestType int
+
+const (
+	CA RequestType = iota + 1
+	KeyPair
+	KeyWithSignedCert
+)
+
 const (
-	SecretCertName = "tls.crt"
-	SecretKeyName  = "tls.key"
+	SecretPrivateKey = "private"
+	SecretPublicKey  = "public"
```

Comment:
Aren't these names nonstandard? I've always seen tls.crt and tls.key in pki secrets.

Reply:
Secret type used is Opaque so don't need to use `tls.crt` and `tls.key`. By using opaque, I can use the same object type for secrets and kubeconfigs in secrets.